File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraPrinting;
using DevExpress.XtraPivotGrid;
using DevExpress.XtraPivotGrid.Data;
using System.Drawing.Printing;
using DevExpress.Drawing.Printing;

namespace S130362 {
	public partial class Form1 : Form {
		public Form1() {
			InitializeComponent();			
		}

        private void Form1_Load(object sender, EventArgs e) {
            sqlDataSource1.Fill();
        }

        private void button1_Click(object sender, EventArgs e) {
			PrintingSystem ps = new PrintingSystem();
			PrintableComponentLink link = new PrintableComponentLink(ps);
			link.Component = pivotGridControl1;
			PivotGridPageSettings pageSettings = pivotGridControl1.OptionsPrint.PageSettings;
			link.PaperKind = (DXPaperKind)pageSettings.PaperKind;
			if(pageSettings.PaperWidth != 0 && pageSettings.PaperHeight != 0)
				link.CustomPaperSize = new Size(pageSettings.PaperWidth, pageSettings.PaperHeight);
			link.Landscape = pageSettings.Landscape;
			link.Margins = pageSettings.Margins;

			link.ShowPreviewDialog(this);

			PivotGridPageSettings pivotSettings = pivotGridControl1.OptionsPrint.PageSettings;
			pivotSettings.PaperKind = (PaperKind)link.PaperKind;
			if(pivotSettings.PaperKind == PaperKind.Custom) {
				pivotSettings.PaperWidth = link.CustomPaperSize.Width;
				pivotSettings.PaperHeight = link.CustomPaperSize.Height;
			}
			pivotSettings.Landscape = link.Landscape;
			pivotSettings.Margins = link.Margins;

			link.Dispose();
			ps.Dispose();
		}

		
	}
}

File: VB/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraPrinting
Imports DevExpress.XtraPivotGrid
Imports System.Drawing.Printing

Namespace S130362

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            sqlDataSource1.Fill()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs)
            Dim ps As PrintingSystem = New PrintingSystem()
            Dim link As PrintableComponentLink = New PrintableComponentLink(ps)
            link.Component = pivotGridControl1
            Dim pageSettings As PivotGridPageSettings = pivotGridControl1.OptionsPrint.PageSettings
            link.PaperKind = pageSettings.PaperKind
            If pageSettings.PaperWidth <> 0 AndAlso pageSettings.PaperHeight <> 0 Then link.CustomPaperSize = New Size(pageSettings.PaperWidth, pageSettings.PaperHeight)
            link.Landscape = pageSettings.Landscape
            link.Margins = pageSettings.Margins
            link.ShowPreviewDialog(Me)
            Dim pivotSettings As PivotGridPageSettings = pivotGridControl1.OptionsPrint.PageSettings
            pivotSettings.PaperKind = link.PaperKind
            If pivotSettings.PaperKind = PaperKind.Custom Then
                pivotSettings.PaperWidth = link.CustomPaperSize.Width
                pivotSettings.PaperHeight = link.CustomPaperSize.Height
            End If

            pivotSettings.Landscape = link.Landscape
            pivotSettings.Margins = link.Margins
            link.Dispose()
            ps.Dispose()
        End Sub
    End Class
End Namespace
